solvers/qpSpline/PiecewisePolynomial.cpp

```diff
@@ -0,0 +1,116 @@
+#include "PiecewisePolynomial.h"
+#include <stdexcept>
+#include <cassert>
+#include <algorithm>
+
+using namespace std;
+
+PiecewisePolynomial::PiecewisePolynomial(std::vector<Polynomial> const& polynomials, std::vector<double> const& segment_times) :
```

Comment:
again, this is almost a mirror of our PPTrajectory class in matlab.  Making those two match would make this much more useful in general.  also, instead of solvers, it might be a better fit to put some of this in systems/trajectories